_includes/guide_p2p_network.md

```diff
@@ -140,6 +140,180 @@ In order to maintain a connection with a peer, nodes by default will send a mess
+
+![First Inv Message Sent During IBD](/img/dev/en-ibd-inv.svg)
+
+Inventories are unique identifiers for information on the network. Each
+inventory contains a type field and the unique identifier for an
+instance of the object. For blocks, the unique identifier is a hash of
+the block's header.
+
+The block inventories appear in the `inv` message in the same order they
+appear in the block chain, so this first `inv` message contains
+inventories for blocks 1 through 501. (For example, the hash of block 1
+is 4860...0000 as seen in the illustration above.)
+
+The IBD node uses the received inventories to request 128 blocks from
+the sync node in the `getdata` message illustrated below.
+
+![First GetData Message Sent During IBD](/img/dev/en-ibd-getdata.svg)
+
+It's important to headers-first nodes that the blocks be requested and
+sent in order because each block header references the header hash of
+the preceeding block. That means the IBD node can't fully validate a
+block until its parent block has been received. Blocks that can't be
+validated because their parents haven't been received are called orphan
+blocks; a subsection below describes them in more detail.
+
+Upon receipt of the `getdata` message, the sync node replies with each
+of the blocks requested. Each block is put into serialized block format
+and sent in a separate `block` message. The first `block` message sent
+(for block 1) is illustrated below.
+
+![First Block Message Sent During IBD](/img/dev/en-ibd-block.svg)
+
+The IBD node downloads each block, validates it, and then requests the
+next block it hasn't requested yet, maintaining a queue of up to 128
+blocks to download. When it has requested every block for which it has
+an inventory, it sends another `getblocks` message to the sync node
+requesting the inventories of up to 500 more blocks.  This second
+`getblocks` message contains multiple header hashes as illustrated
+below:
+
+![Second GetBlocks Message Sent During IBD](/img/dev/en-ibd-getblocks2.svg)
+
+Upon receipt of the second `getblocks` message, the sync node takes the
+first listed header hash and searches its local best block chain for a
+block with that header hash. If it finds a block with that hash, it
+replies with 500 block inventories starting with the following block.
+But if it doesn't find a block with that hash, it takes the next header
+hash from the `getblocks` message and searches its block chain for that
+hash. If that hash matches, it will reply with 500 block inventories
+starting with the following hash from that point. But, again, if it
+doesn't find that hash, it will proceed to check the next hash in the
+message (and so on until it runs out of hashes in the message). If the
+last hash in the message (besides the stopping hash) doesn't match, it
+assumes the only block the two nodes have in common is block 0 and so it
+sends an `inv` starting with block 1 (the same `inv` message seen
+several illustrations above).
```

Comment:
@harding FWIW, I was temporarily confused here, probably due to the repeating and conditional sentences, and felt this could perhaps be explained in a more concise way. What would you think of something similar to the following?

----

Upon receipt of the second getblocks message, the sync node searches its local best block chain for a block that matches one of the header hashes in the message, trying each hash in the order they were received. As soon as it finds a matching hash, it will reply with 500 block inventories starting with the following hash from that point. However if there is no matching hash (besides the stopping hash), it assumes the only block the two nodes have in common is block 0 and so it sends an inv starting with block 1 (the same inv message seen several illustrations above).